CONTRIBUTING.md

```diff
@@ -0,0 +1,183 @@
+# Framework Philosophies
+
+This document outlines the goals, ideas, and roadmap for the rg-bot library (as well as our public-facing libraries in general). This is a living document, and we encourage players to participate and contribute to this document.
+
+## Core Facts
+
+The success of this library is dictated by its adherence to several core facts. Our goal is to make sure that these facts inform changes to the library. Any change that violates one or more of these facts must be re-evaluated.
+
+Note that some of these facts require testing the library against real users. We don't need to evaluate these specific items every time a change occurs in the library, but we should make that evaluation often.
+
+#### Fact 1: 80% of users with more than 6 months and less than 1 year of JavaScript experience should be able to articulate what a provided example / template does.
+Making this goal a reality ensures that most new players who may not be coding experts can jump right in and begin understanding the bots on the Regression Games platform. This is measured by conducting user-interviews and surveys with potential players. The user should be able to _roughly describe_ the algorithm in the example (we define this as getting most of the algorithm). Here is an example of a template:
+
+```javascript
+const animalsToHunt = ["chicken", "pig", "cow", "sheep", "player"];
+
+async function huntAnimals() {
+    let nearbyAnimals = bot.findEntities({entityNames: animalsToHunt, maxDistance: 100});
+    console.log(nearbyAnimals)
+    if (nearbyAnimals.length == 0) {
+        bot.chat("Could not find animals nearby... try wandering around first with `bot.wander()`")
+        return
+    }
+    let animalToAttack = nearbyAnimals[0].result
+    bot.chat("Hunting a " + animalToAttack.name)
+    let attackCount = 0
+    while (animalToAttack.isValid) 
+        if (attackCount > 100) {
+            bot.chat("Could not kill animal, stopping. Try hunting a new animal.")
+            return;
+        }
+        await bot.attackEntity(animalToAttack)
+    }
+    bot.chat("Finished attacking the " + animalToAttack.type, ", moving on the next victim")
+    let itemsOnGround = await bot.findAndCollectItemsOnGround()
+    bot.chat(`Picked up ${itemsOnGround.length} items off the ground`)
+    return await huntAnimals()
```

Comment:
I would suggest using the code sample , or very close to it, from the replit example.

https://linear.app/regression-games/issue/REG-505#comment-cdf59209